### PR TITLE
WPT variant name should not be an empty string

### PR DESCRIPTION
**Pull request by WeizhongX**
Change existing document and tests to not use empty string for variant

Bug: 40952

### CODE REVIEW

File: tools/lint/rules.py [hunk 1]
```diff
@@ -198,8 +198,8 @@ class VariantMissing(Rule):
 class MalformedVariant(Rule):
     name = "MALFORMED-VARIANT"
     description = collapse("""
-        %s `<meta name=variant>` 'content' attribute must be the empty string
-        or start with '?' or '#'
+        %s `<meta name=variant>` 'content' attribute must be a non empty string
+        and starts with '?' or '#'
```
Comment (jgraham) [inline tools/lint/rules.py:202]:
```suggestion
        and start with '?' or '#'
```

**Review by jgraham — COMMENTED** [review]
So I'm confused. I thought we had ~200 tests with empty variant names, but this PR doesn't seem to fix that many, but passes the lint rule. So are the lint rules missing something?

File: tools/lint/rules.py (hunk 1 above)
Comment (WeizhongX) [reply tools/lint/rules.py]: updated, thanks!

**Review by jonathan-j-lee — APPROVED** [review]
LGTM, but we need for the comment period on web-platform-tests/rfcs#158 to pass.

**Comment by WeizhongX** [comment]
> LGTM, but we need for the comment period on [web-platform-tests/rfcs#158](https://github.com/web-platform-tests/rfcs/pull/158) to pass.

We had a meeting discuss RFC 158, and it is approved, though not merged yet.

This one is not super urgent. Only risk is that someone can land a new test with empty variant.

**Pull request merged**